tests/pytests/unit/states/test_jboss7.py

```diff
@@ -0,0 +1,748 @@
+        },
+    ):
+        # when
+        result = jboss7.deployed(
+            name="unchanged", jboss_config=jboss_conf, salt_source=parameters
+        )
+
+        # then
+        assert not jboss7_undeploy_mock.called
+        assert not jboss7_deploy_mock.called
+
+
+def test_should_redeploy_changed():
+    # given
+    parameters = {
+        "target_file": "some_artifact",
+        "undeploy_force": False,
+        "undeploy": "some_artifact",
+        "source": "some_artifact_on_master",
+    }
+    jboss_conf = {"cli_path": "somewhere", "controller": "some_controller"}
+
+    def list_deployments(jboss_config):
+        return ["some_artifact"]
+
+    def file_get_managed(
+        name,
+        template,
+        source,
+        source_hash,
+        source_hash_name,
+        user,
+        group,
+        mode,
+        attrs,
+        saltenv,
+        context,
+        defaults,
+        skip_verify,
+        kwargs,
+    ):
+        return "sfn", "hash", ""
+
+    def file_manage_file(
+        name,
+        sfn,
+        ret,
+        source,
+        source_sum,
+        user,
+        group,
+        mode,
+        attrs,
+        saltenv,
+        backup,
+        makedirs,
+        template,
+        show_diff,
+        contents,
+        dir_mode,
+    ):
+        return {"result": True, "changes": True}
+
+    jboss7_undeploy_mock = MagicMock()
+    jboss7_deploy_mock = MagicMock()
+    file_get_managed = MagicMock(side_effect=file_get_managed)
+    file_manage_file = MagicMock(side_effect=file_manage_file)
+    list_deployments_mock = MagicMock(side_effect=list_deployments)
+    with patch.dict(
+        jboss7.__salt__,
+        {
+            "jboss7.undeploy": jboss7_undeploy_mock,
+            "jboss7.deploy": jboss7_deploy_mock,
+            "file.get_managed": file_get_managed,
+            "file.manage_file": file_manage_file,
+            "jboss7.list_deployments": list_deployments_mock,
+        },
+    ):
+        # when
+        result = jboss7.deployed(
+            name="unchanged", jboss_config=jboss_conf, salt_source=parameters
+        )
+
+        # then
+        assert jboss7_undeploy_mock.called
+        assert jboss7_deploy_mock.called
+
+
+def test_should_deploy_different_artifact():
+    # given
+    parameters = {
+        "target_file": "some_artifact",
+        "undeploy_force": False,
+        "undeploy": "some_artifact",
+        "source": "some_artifact_on_master",
+    }
+    jboss_conf = {"cli_path": "somewhere", "controller": "some_controller"}
+
+    def list_deployments(jboss_config):
+        return ["some_other_artifact"]
+
+    def file_get_managed(
+        name,
+        template,
+        source,
+        source_hash,
+        source_hash_name,
+        user,
+        group,
+        mode,
+        attrs,
+        saltenv,
+        context,
+        defaults,
+        skip_verify,
+        kwargs,
+    ):
+        return "sfn", "hash", ""
+
+    def file_manage_file(
```

Comment:
instead of defining these functions, cant we just mock the return? It doesn't look like these functionins throughout this test case are doing much other than returning something.

Reply:
Fixed, please note I intentionally left some inner functions as they all provide different logic so we might end up with same number of fixtures.